vic/drivers/image/src/vic_write.c

```diff
@@ -35,26 +35,29 @@
 void
 vic_write(void)
 {
-    extern out_data_struct **out_data;
-    extern domain_struct     global_domain;
-    extern domain_struct     local_domain;
-    extern nc_file_struct    nc_hist_file;
-    extern nc_var_struct     nc_vars[N_OUTVAR_TYPES];
-    extern size_t            current;
-    int                      dimids[MAXDIMS];
-    size_t                   i;
-    size_t                   j;
-    size_t                   k;
-    size_t                   grid_size;
-    size_t                   ndims;
-    double                  *dvar = NULL;
-    size_t                   dcount[MAXDIMS];
-    size_t                   dstart[MAXDIMS];
+    extern out_data_struct   **out_data;
+    extern domain_struct       global_domain;
+    extern domain_struct       local_domain;
+    extern nc_file_struct      nc_hist_file;
+    extern nc_var_struct       nc_vars[N_OUTVAR_TYPES];
+    extern size_t              current;
+    extern global_param_struct global_param;
+    extern dmy_struct         *dmy;
+    int                        dimids[MAXDIMS];
+    size_t                     i;
+    size_t                     j;
+    size_t                     k;
+    size_t                     grid_size;
+    size_t                     ndims;
+    double                    *dvar = NULL;
+    size_t                     dcount[MAXDIMS];
+    size_t                     dstart[MAXDIMS];
+    int                        nc_time;
 
     grid_size = global_domain.n_ny * global_domain.n_nx;
 
     // allocate memory for variables to be stored
-    dvar = (double *) malloc(local_domain.ncells * sizeof(double));
+    dvar = calloc(local_domain.ncells, sizeof(*dvar));
```

Comment:
Yes, but it was a mistake of me while adapting the cast syntax you mentioned. I am used to use calloc in stead of malloc (calloc is a bit slower because it it zeroing out memory is'nt it?). anyway I changed it to:
```
    dvar = malloc(local_domain.ncells * sizeof(*dvar));
```